code/dashboard-2/src/util/query/QueryUtils.ts

```diff
@@ -155,47 +155,47 @@ export function compileQuery(query: string, knownFields: Record<string, string |
     return false
   }
 
-  function exprBin(next, constructor, ...ops) {
+  function exprBin(next: any, constructor: any, ...ops : any[]) {
     return function () {
       let e = next()
       Outer:
-        for (; ;) {
-          for (let op of ops) {
-            if (eatToken(op)) {
-              let e2 = next()
-              try {
-                e = new constructor(op, e, e2)
-              } catch (ex) {
-                fail(ex)
-              }
-              continue Outer
+      for (; ;) {
+        for (const op of ops) {
+          if (eatToken(op)) {
+            const e2 = next()
+            try {
+              e = new constructor(op, e, e2)
+            } catch (ex) {
+              fail(ex)
             }
+            continue Outer
           }
-          break
         }
+        break
+      }
       return e
     }
   }
 
   function exprPrim() {
     if (eatToken('(')) {
-      let e = exprOr()
+      const e = exprOr()
       if (!eatToken(')')) {
         fail('Expected \')\' here')
       }
       return e
     }
     if (eatToken('~')) {
-      let e = exprPrim()
+      const e: unknown = exprPrim()
```

Comment:
Does the type annotation add any information that is useful, and if it does, why is that type of annotation not used more broadly?  And why "unknown"?  At worst this should be "any" (which you could leave off unless your linter is particularly fascistic), "unknown" doesn't seem to buy you anything.  And NotOperation takes an "any" parameter.

(I don't much care for the transition from "let" to "const" here b/c I don't think the intent of the bindings is to make something constant; it is happenstance that the variable is not updated.  But if you like it better that way, go ahead.)

Reply:
I used `unknown` in the first place because the linter suggested to use `unknown` instead of `any`.
https://typescript-eslint.io/rules/no-explicit-any/

However I added a new type and assigned it. Hope it will be ok.